rice-middleware/impl/src/main/java/org/kuali/rice/kew/actions/ReturnToPreviousNodeAction.java

```diff
@@ -382,10 +382,21 @@ private void assertValidProcess(RouteNodeInstance resultNodeInstance, Collection
     private void assertFinalApprovalNodeNotInPath(List path) throws InvalidActionTakenException {
     	for (Iterator iterator = path.iterator(); iterator.hasNext(); ) {
 			RouteNodeInstance  nodeInstance = (RouteNodeInstance ) iterator.next();
-			// if we have a complete final approval node in our path, we cannot return past it
-			if (nodeInstance.isComplete() && Boolean.TRUE.equals(nodeInstance.getRouteNode().getFinalApprovalInd())) {
-				throw new InvalidActionTakenException("Cannot return past or through the final approval node '"+nodeInstance.getName()+"'.");
-			}
+            List<ActionTakenValue> actionsTaken = KEWServiceLocator.getActionTakenService().getActionsTakenAtRouteNode(nodeInstance);
+            ActionTakenValue foundValue =  null;
+            String actionTknCode = null;

```

Comment:
We rather spell out the variable name actionTakenCode, no need to abbreviate it here.  In this specific case I'd just use `actionTaken` to stay consistent with the naming in ActionTakenValue object.